amundsen_application/static/js/components/common/ScrollTracker/index.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { logAction } from "../../../ducks/utilMethods";
+
+
+export interface ScrollTrackerProps {
+  location: string;
+  targetId: string;
+}
+
+interface ScrollTrackerState {
+  thresholds: number[];
+}
+
+export default class ScrollTracker extends React.Component<ScrollTrackerProps, ScrollTrackerState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      thresholds: [25, 50, 75, 100]
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", this.onScroll);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("scroll", this.onScroll)
+  }
+
+  // TODO - Debounce this function
```

Comment:
How do you feel about including lodash into Amundsen? Would be a great utility add.

Reply:
Up for doing that in a separate PR. Main concern is handling that before we deploy this code so we don't start firing large amounts of scroll logging events in production.

Reply:
I tested the scroll events locally and there shouldn't be any performance concerns at this point. I also clean up the events so they don't stack on top of each other as you continue to use the site.

Reply:
Alternatively we could just use 'debounce' which was taken from underscore's codebase: https://www.npmjs.com/package/debounce

Reply:
I'm not concerned about page performance so much thinking about whether or not we risk bloating our production logs. Out of curiosity what's the general magnitude of events getting fired when exploring a page for let's say a minute?

Reply:
If the user scrolls all the way to the bottom of the table details page, a total of 4 analytics events will be fired (25, 50, 75, and 100). These events do not get sent multiple times because of the `shift()` on line 44.

Reply:
I see. This becomes more of a nit then, doesn't look like we'll blow up our production logs and we can get around to the debouncing when there's spare time.

Reply:
Added throttling.